zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +133,137 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() { // Test multiple small transactions to fill a block max size
+    let blockheader = generate_block_header();
+
+    // 23511 transactions = data.len() = 1999925 bytes
+    // 23512 transactions = data.len() = 2000010 bytes
+    let mut limit_txs = 23511;
+
+    // Create 23511 transactions, this is the limit with our DUMMY_TX1 data
+    let mut many_transactions = Vec::new();
+    for _n in 1..limit_txs+1 {
+        let trx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]);
+        many_transactions.push(Arc::new(trx.unwrap()));
+    }
```

Comment:
TIOLI: you might be able to simplify this a little with `collect` and `repeat`

```rust
let many_transactions = std::iter::repeat(Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap())
    .take(limit_txs)
    .collect::<Vec<_>>();
```

Reply:
Thanks, applied.